src/diffusers/models/prior_transformer.py

```diff
@@ -0,0 +1,146 @@
+from dataclasses import dataclass
+from typing import Union
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from ..configuration_utils import ConfigMixin, register_to_config
+from ..modeling_utils import ModelMixin
+from ..utils import BaseOutput
+from .attention import BasicTransformerBlock
+from .embeddings import TimestepEmbedding, Timesteps
+
+
+@dataclass
+class PriorTransformerOutput(BaseOutput):
+    predicted_image_embedding: torch.FloatTensor
+
+
+class PriorTransformer(ModelMixin, ConfigMixin):
+    @register_to_config
+    def __init__(
+        self,
+        num_attention_heads: int = 32,
+        attention_head_dim: int = 64,
+        num_layers: int = 20,
+        clip_embeddings_dim: int = 768,
+        clip_num_embeddings=77,
+        additional_embeddings=4,
+        dropout: float = 0.0,
+        upcast_attention: bool = False,
+    ):
+        super().__init__()
+        self.num_attention_heads = num_attention_heads
+        self.attention_head_dim = attention_head_dim
+        inner_dim = num_attention_heads * attention_head_dim
+        self.additional_embeddings = additional_embeddings
+
+        self.time_proj = Timesteps(inner_dim, True, 0)
+        self.time_embedding = TimestepEmbedding(inner_dim, inner_dim)
+
+        self.proj_in = nn.Linear(clip_embeddings_dim, inner_dim)
+
+        self.text_embeddings_proj = nn.Linear(clip_embeddings_dim, inner_dim)
+        self.text_encoder_hidden_states_proj = nn.Linear(clip_embeddings_dim, inner_dim)
+
+        self.positional_embedding = nn.Parameter(
+            torch.zeros(1, clip_num_embeddings + additional_embeddings, inner_dim)
+        )
+
+        # TODO - better name. I can't tell what this is
+        self.prd_embedding = nn.Parameter(torch.zeros(1, 1, inner_dim))
```

Comment:
yes can we maybe find a better name here?

Reply:
pinged kakaobrain in slack

Reply:
actually not that big of a deal - let's just leave it